### PR TITLE
"About This Guide" updates following Guide reorganization

### PR DESCRIPTION
**Pull request by chrissimpkins**
This PR updates the About This Guide page with the new section titles/descriptions following the reorganization in #651.

### CODE REVIEW

File: src/part-2-intro.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Part 2: How rustc works
+# Part 2: How `rustc` Works
```
Comment (mark-i-m) [inline src/part-2-intro.md:1]: Hmm... I think high-level compiler arch is actually a better name (also in the about this guode section).

File: src/about-this-guide.md [hunk 2]
```diff
@@ -6,16 +6,16 @@ development.
...
-   representations.
-4. [The Type System][p4]: Discusses the type system.
-5. [The Compiler Backend][p5]: Discusses the compiler backend, code generation,
-   linking, and debug info.
+2. [How `rustc` Works][p2]: Discusses the high-level
+   architecture of the compiler and stages of the compile process
+3. [Source Code Representation][p3]: Describes the process of taking raw source code from the user and
+   transforming it into various forms that the compiler can work with easily
+4. [Analysis][p4]: discusses the analyses that the compiler uses to check various
+   properties of the code and inform later stages of the compile process
```
Comment (mark-i-m) [inline src/about-this-guide.md:17]: perhaps add "(e.g., type checking)”

File: src/about-this-guide.md [hunk 3]
```diff
@@ -6,16 +6,16 @@ development.
...
+5. [From MIR to Binaries][p5]: How linked executable machine code is generated
```
Comment (mark-i-m) [inline src/about-this-guide.md:18]: Minor nit: can you end each point with a period?

Comment (chrissimpkins) [reply src/about-this-guide.md:18]: No problem!  Will add them now.

File: src/part-2-intro.md (hunk 1 above)
Comment (chrissimpkins) [reply src/part-2-intro.md]: 978d894938f0d0990df1b82c4b72ee77381625f6

File: src/about-this-guide.md (hunk 2 above)
Comment (chrissimpkins) [reply src/about-this-guide.md]: 978d894938f0d0990df1b82c4b72ee77381625f6

File: src/about-this-guide.md (hunk 3 above)
Comment (chrissimpkins) [reply src/about-this-guide.md]: 978d894938f0d0990df1b82c4b72ee77381625f6

**Pull request merged**